co-simulation tool/bridge/carla_mosaic_bridge.py

```diff
@@ -0,0 +1,172 @@
+
+if 'SUMO_HOME' in os.environ:
+    sys.path.append(os.path.join(os.environ['SUMO_HOME'], 'tools'))
+else:
+    sys.exit("please declare environment variable 'SUMO_HOME'")
+
+# ==================================================================================================
+# -- imports ---------------------------------------------------------------------------------------
+# ==================================================================================================
+
+import sumolib  # pylint: disable=wrong-import-position
+import traci  # pylint: disable=wrong-import-position
+
+from carla_integration.carla_simulation import CarlaSimulation  # pylint: disable=wrong-import-position
+from carla_integration.sumo_simulation import SumoSimulation  # pylint: disable=wrong-import-position
+
+from carla_integration.synchronization import SimulationSynchronization  # pylint: disable=wrong-import-position
+
+from util.netconvert_carla import netconvert_carla
+# ==================================================================================================
+# -- main ------------------------------------------------------------------------------------------
+# ==================================================================================================
+
+
+def main(args):
+
+    # ----------------
+    # carla simulation
+    # ----------------
+    carla_simulation = CarlaSimulation(args.host, args.port, args.step_length)
+    if args.map is not None:
+        print('load map %r.' % args.map)
+        world = carla_simulation.client.load_world(args.map)
+    else:
+        world = carla_simulation.client.get_world()
+
+    # ---------------
+    # sumo simulation
+    # ---------------
+    if args.net_file is not None:
+        sumo_net=sumolib.net.readNet(args.net_file)
+        print("load net file")
+    else:
+        # Temporal folder to save intermediate files.
+        tmpdir = tempfile.mkdtemp()    
+        current_map = world.get_map()
+        xodr_file = os.path.join(tmpdir, current_map.name + '.xodr')
+        current_map.save_to_disk(xodr_file)
+        net_file = os.path.join(tmpdir, current_map.name + '.net.xml')
+        netconvert_carla(xodr_file, net_file, guess_tls=True)
+        sumo_net = sumolib.net.readNet(net_file)
+
+    sumo_simulation = SumoSimulation(sumo_net,
+                                     args.step_length,
+                                     host=args.bridge_server_host,
+                                     port=args.bridge_server_port)
+
+    # ---------------
+    # synchronization
+    # ---------------
+    synchronization = SimulationSynchronization(sumo_simulation, carla_simulation, args.tls_manager,
+                                                args.sync_vehicle_color, args.sync_vehicle_lights)
+
+    # start simulation synchronization
+    try:
+        while True:
+            synchronization.tick()
```

Comment:
There is an exception occurring at the end of the program.

Reply:
The exception has been handled by adding "try" and "except" methods.